test/GPv2Settlement/ComputeTradeExecutions.t.sol

```diff
@@ -0,0 +1,185 @@
+// SPDX-License-Identifier: LGPL-3.0-or-later
+pragma solidity ^0.8.26;
+
+import {Vm} from "forge-std/Test.sol";
+
+import {GPv2Settlement} from "src/contracts/GPv2Settlement.sol";
+import {GPv2Order, IERC20} from "src/contracts/libraries/GPv2Order.sol";
+import {GPv2Signing} from "src/contracts/mixins/GPv2Signing.sol";
+
+import {GPv2Transfer, Helper, SettlementEncoder} from "./Helper.sol";
+
+import {Order} from "test/libraries/Order.sol";
+import {Registry, TokenRegistry} from "test/libraries/encoders/TokenRegistry.sol";
+
+abstract contract BaseComputeTradeExecutions is Helper {
+    using TokenRegistry for TokenRegistry.State;
+    using TokenRegistry for Registry;
+    using SettlementEncoder for SettlementEncoder.State;
+
+    IERC20 internal sellToken;
+    IERC20 internal buyToken;
+
+    uint256 internal executedAmount = 10 ether;
+    uint256 internal sellPrice = 1;
+    uint256 internal buyPrice = 2;
+
+    function setUp() public virtual override {
+        super.setUp();
+
+        sellToken = IERC20(makeAddr("GPv2Settlement.BaseComputeTradeExecutions: sellToken"));
+        buyToken = IERC20(makeAddr("GPv2Settlement.BaseComputeTradeExecutions: buyToken"));
+    }
+
+    function defaultOrder() internal view returns (GPv2Order.Data memory) {
+        return GPv2Order.Data({
+            sellToken: sellToken,
+            buyToken: buyToken,
+            receiver: GPv2Order.RECEIVER_SAME_AS_OWNER,
+            sellAmount: 42 ether,
+            buyAmount: 13.37 ether,
+            validTo: type(uint32).max,
+            appData: bytes32(0),
+            feeAmount: 0,
+            kind: bytes32(0),
+            partiallyFillable: false,
+            sellTokenBalance: GPv2Order.BALANCE_ERC20,
+            buyTokenBalance: GPv2Order.BALANCE_ERC20
+        });
+    }
+
+    function computeSettlementForOrder(GPv2Order.Data memory order)
+        internal
+        returns (uint256 executedSellAmount, uint256 executedBuyAmount)
+    {
+        encoder.tokenRegistry.tokenRegistry().setPrice(sellToken, sellPrice);
+        encoder.tokenRegistry.tokenRegistry().setPrice(buyToken, buyPrice);
+        encoder.signEncodeTrade(vm, trader, order, domainSeparator, GPv2Signing.Scheme.Eip712, executedAmount);
+
+        SettlementEncoder.EncodedSettlement memory encoded = encoder.encode(settlement);
+        (GPv2Transfer.Data[] memory inTransfers, GPv2Transfer.Data[] memory outTransfers) =
+            settlement.computeTradeExecutionsTest(encoded.tokens, encoded.clearingPrices, encoded.trades);
+
+        // As `vm.expectRevert` may be used prior to this function, `computeTradeExecutionsTest` may revert however
+        // execution within this function will still continue. This is a workaround to prevent the panic.
+        if (inTransfers.length != 0 && outTransfers.length != 0) {
```

Comment:
If even one of the two is nonzero it means that the transaction didn't revert and an out-of-bound index is desired behavior.

```suggestion
        // As `vm.expectRevert` may be used prior to this function, `computeTradeExecutionsTest` may revert however
        // execution within this function will still continue. This is a workaround to prevent the panic.
        if (inTransfers.length != 0 || outTransfers.length != 0) {
```